### ISSUE TITLE
Support flashing over USB

### ISSUE THREAD

**Issue by technobly** [spark/spark-dev#6]
I would love to see the Spark DFU-UTIL package installed by default.

Also, I had DFU UTIL working in 0.0.13.  Installed 0.0.14 just now and I'm getting this error (below).. if I delete the ~/.sparkide/ the dfu util is gone and I get other errors trying to search for it in the package manager.  If I reinstall it with 0.0.13 and then open 0.0.14 I get this error below...
http://i.imgur.com/BLTcH5J.png

**Comment by mfogel** [comment]
Should this thread be renamed to something like 'Get flash via USB working in Particle Dev'? Or else it should probably be closed... seems a decision was made and [implemented in the code](https://github.com/spark/particle-dev/blob/v0.1.5/lib/main.coffee#L731) to not install the dfu-util dependencies by default on account of license issues. That covers this issue as described, but the larger issue is that flashing via USB from particle dev doesn't currently work. Even worse... it silently fails without a warning or error message because of [this pass through](https://github.com/spark/particle-dev/blob/v0.1.5/lib/main.coffee#L733).

In terms of getting that to work, it appears the [spark-dfu-util](https://github.com/suda/spark-dfu-util) package (hosted in @suda 's account) needs a naming update to 'particle-dev-dfu-util' so the [call from the particle-dev package](https://github.com/spark/particle-dev/blob/v0.1.5/lib/main.coffee#L735) will work correctly. I assume there's more issues than that as spark-dfu-util hasn't been updated in over 2 years now but that's the first obvious one.

**Comment by suda** [comment]
Good point Mike! It would be possible to have flashing via USB without dfu-util and use ymodem protocol. I renamed this issue to reflect the nature of the issue better.
